ros/src/tl_detector/tl_detector.py

```diff
@@ -115,12 +130,29 @@ def get_light_state(self, light):
         """
         if(not self.has_image):
             self.prev_light_loc = None
-            return False
+            return TrafficLight.UNKNOWN
 
         cv_image = self.bridge.imgmsg_to_cv2(self.camera_image, "bgr8")
 
         #Get classification
-        return self.light_classifier.get_classification(cv_image)
+        #return self.light_classifier.get_classification(cv_image)
+	return light.state
+
+    def is_ahead_of(self, pose, x, y):
+        """Determines whether a wappoint is ahead of position
+        Args:
+            pose: base position
+            x (float): waypoint's global x-coordinate
+            y (float): waypoint's global y-coordinate
+        Returns:
+            bool: Whether the waypoint is ahead of this position
+        """
+        x1 = pose.position.x
+        y1 = pose.position.y
+        orientation = pose.orientation
+        euler = tf.transformations.euler_from_quaternion([orientation.x, orientation.y, orientation.z, orientation.w])
+        yaw = euler[2]
+        return ((x - x1) * math.cos(yaw) + (y - y1) * math.sin(yaw)) > 0
```

Comment:
Looks good. Another approach would be compute the heading with respect to map coords between the car and the waypoint. Then check that the abs (yaw - heading) is less the pi/4.

Reply:
Yes, that would work too. Let's keep it at this stage.